src/BudgetManager/BudgetManager.CQRS/Handlers/CategoryHandlers/AddCategoryHandler.cs

```diff
@@ -4,25 +4,35 @@
 using BudgetManager.SDK.DTOs.CategoryDTOs;
 using BudgetManager.Shared.DataAccess.MongoDB.BaseImplementation;
 using MediatR;
+using MongoDB.Driver;
 
 namespace BudgetManager.CQRS.Handlers.CategoryHandlers
 {
     public class AddCategoryHandler : IRequestHandler<AddCategoryCommand, Guid>
     {
-        private readonly IBaseRepository<Category> _categoryRepository;
+        private readonly IBaseRepository<User> _userRepository;
         private readonly IMapper _mapper;
 
-        public AddCategoryHandler(IBaseRepository<Category> categoryRepository, IMapper mapper)
+        public AddCategoryHandler(IBaseRepository<User> userRepository, IMapper mapper)
         {
-            _categoryRepository = categoryRepository;
+            _userRepository = userRepository;
             _mapper = mapper;
         }
 
         public async Task<Guid> Handle(AddCategoryCommand request, CancellationToken cancellationToken)
         {
+            Guid userId = request.category.UserId;
             AddCategoryDTO requestCategory = request.category;
+            var user = await _userRepository.FindByIdAsync(userId, cancellationToken);
+
+            if (user is null) { return Guid.Empty; }
+
             Category mappedCategory = _mapper.Map<Category>(requestCategory);
-            await _categoryRepository.InsertOneAsync(mappedCategory, cancellationToken);
+
+            var filter = Builders<User>.Filter.Eq(u => u.Id, userId);
+            var update = Builders<User>.Update.Push(u => u.Categories, mappedCategory);
+
+            await _userRepository.UpdateOneAsync(filter, update, cancellationToken);
```

Comment:
No need for lines 26-29, just check whether like 35 returns a user with a valid Guid